openstack_plugin/resources/compute/server.py

```diff
@@ -1252,6 +1357,21 @@ def _handle_detach_external_volume_from_server():
                     'external volume and server are being used')
 
 
+def _handle_attach_security_group_to_server(openstack_resource,
+                                            security_groups):
+    """
+    This method will connect all security groups configured to server using
+    relationships or node properties to attach them to
+    :param openstack_resource: instance of openstack server resource
+    :param security_groups: List of security groups which should be
+    added to the server
+    """
+    for security_group in security_groups or []:
+        openstack_resource.add_security_group_to_server(
+            security_group.get('name')
```

Comment:
I assume this can take either id or name?

Reply:
Yes, based on the Openstack guy I talked to, this can be a name or id,  passing name has a major bug so we place the id value and get it from the name attr